src/search3_response.erl

```diff
@@ -131,6 +132,12 @@ handle_response({ok, #{session := Session} = Response, _},
 handle_response({error, Error}, _) ->
     handle_error_response({error, Error}).
 
+handle_update_response({ok, #{session := Session}, _}, CurrentSession) ->
+    VSession = verify_same_session(CurrentSession, Session),
+    VSession;
```

Comment:
this doesn't look like the right result. handle_response returns `{VSession, Response}` above, for example.

Reply:
@rnewson `handle_response` was a generic handler and `Response` was need for other rpc requests. However in the case of updates, `Response` isn't needed, so I removed it.